### PR TITLE
fix: initialize the user source using the client internalID and not t…

### PR DESCRIPTION
**Pull request by leleueri**
…he client_id

fixes AM-2166

gravitee-io/issues#9500

### CODE REVIEW

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/handler/user/PasswordPolicyRequestParseHandler.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ public void handle(RoutingContext context) {
         Optional<PasswordSettings> passwordSettings = PasswordSettings.getInstance(client, this.domain);
```
Comment (mstasgravitee) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/handler/user/PasswordPolicyRequestParseHandler.java:55]: In bug, I can see
"Versions affected:
4.1.x
4.2.x"

why is this PR to 4.0.x?

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/handler/user/UserRequestHandler.java [hunk 2]
```diff
@@ -91,14 +93,14 @@ private void doRedirect(HttpServerResponse response, String url) {
...
-    protected io.gravitee.am.model.User convert(MultiMap params) {
+    protected io.gravitee.am.model.User convert(MultiMap params, Client client) {
         io.gravitee.am.model.User user = new io.gravitee.am.model.User();
         user.setUsername(params.get("username"));
         user.setFirstName(params.get("firstName"));
         user.setLastName(params.get("lastName"));
         user.setEmail(params.get("email"));
         user.setPassword(params.get("password"));
-        user.setClient(params.get("client_id"));
+        ofNullable(client).ifPresent(cli -> user.setClient(client.getId()));
```
Comment (mstasgravitee) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/handler/user/UserRequestHandler.java:103]: Can we add a test for it?

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/handler/user/PasswordPolicyRequestParseHandler.java (hunk 1 above)
Comment (leleueri) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/handler/user/PasswordPolicyRequestParseHandler.java:55]: it is also present in 4.0

Comment (mstasgravitee) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/handler/user/PasswordPolicyRequestParseHandler.java:55]: ok, please update Jira and public bug then

**Review by mstasgravitee — APPROVED** [review]

**Pull request merged**